sound/soc/sof/intel/hda.c

```diff
@@ -983,12 +983,13 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 	/* create codec instances */
 	hda_codec_probe_bus(sdev);
 
+err:
 	if (!HDA_IDISP_CODEC(bus->codec_mask))
```

Comment:
The `bus->codec_mask` is determined within `hda_dsp_ctrl_init_chip()` as first thing in hda_init_caps(), if there were IDISP identified the patch will not do what the commit message states, the i915 power will be not turned off?

Reply:
indeed, nice catch @ujfalusi 

I will change the test to 
````c
if (!HDA_IDISP_CODEC(bus->codec_mask) || ret < 0)
    hda_codec_i915_display_power(sdev, false);
````

Reply:
nope, this doesn't work. I need to add a boolean and test it.